docs/DirectManipulation.md

```diff
@@ -0,0 +1,224 @@
+---
+id: direct-manipulation
+title: Direct Manipulation
+layout: docs
+category: Guides
+permalink: docs/direct-manipulation.html
+next: linking-libraries
+---
+
+It is sometimes necessary to make changes directly to a component
+without using state/props to trigger a re-render. When using React in
+the browser for example, you sometimes need to directly modify a DOM
+node. This could be necessary for a variety of reasons, such as
+performance (avoid a render pass, useful with animations) or to simplify
+the API of a component.  Both of these apply to `TouchableOpacity`.
+
+```javascript
+setOpacityTo: function(value) {
+  // Redacted: animation related code
+  this.refs[CHILD_REF].setNativeProps({
+    opacity: value
+  });
+},
+```
+[TouchableOpacity.js](https://github.com/facebook/react-native/blob/master/Libraries/Components/Touchable/TouchableOpacity.js)
+
+This allows us to write the following code and know that the child and
+any of its children will have their opacity updated in response to taps.
+
+```javascript
+<TouchableOpacity onPress={this._handlePress}>
+  <View style={styles.button}>
+    <Text>Press me!</Text>
+  </View>
+<TouchableOpacity>
+```
+
+Let's imagine that `setNativeProps` was not available. One way that we
+might implement it with that constraint is to store the opacity value
+in the state, then update that value whenever `onPress` is fired:
+
+```javascript
+getInitialState() {
+  return { myButtonOpacity: 1, }
+},
+
+render() {
+  return (
+    <TouchableOpacity onPress={() => this.setState({myButtonOpacity: 0.5})}
+                      onPressOut={() => this.setState({myButtonOpacity: 1})}>
+      <View style={[styles.button, {opacity: this.state.myButtonOpacity}]}>
+        <Text>Press me!</Text>
+      </View>
+    </TouchableOpacity>
+  )
+}
+```
+
+This is a lot of work compared to the original example - you need to
+track the opacity value and re-render the entire component each time the
+opacity changes. It's possible that you could simplify this further, but
+it will ultimately be some variation of the same approach.
```

Comment:
The point we should convey is that setNativeProps is a perf optimization, so if "a lot of work" refers to a lot of CPU cycles I agree with that and rephrase as

> This is computationally intensive compared to the original example - React needs to re-render the component hierarchy each time the opacity changes, even though other properties of the view and its children haven't changed. Usually this overhead isn't a concern but when performing continuous animations and responding to gestures, judiciously optimizing your components can improve your animations' fidelity.

If it's referring to a lot of typing code, I don't think setNativeProps is the right technique to reduce that because the program becomes more complex in a bunch of ways (this component should own the opacity value but now it doesn't know it, the source of truth of the opacity value is now stored on the native view, and later re-renders could clobber the effects of setNativeProps as you mentioned at the end of this doc).